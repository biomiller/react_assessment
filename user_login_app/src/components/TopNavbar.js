import React from 'react';

import {
    Navbar,
    NavbarBrand,
    NavItem,
    Nav,
    Button
} from 'reactstrap';

import { Link } from "react-router-dom";


import 'bootstrap/dist/css/bootstrap.css';

export function TopNavbar(props) {
    return (
        <div>
            <Navbar color="light" light expand="md">
                <Link to={`/`}><NavbarBrand>React Assessment</NavbarBrand></Link>
                <NavbarBrand>Logged in: {props.loggedIn}</NavbarBrand>
                <Nav>
                    <NavItem>
                        <Link to={`/Register`}><Button>Register</Button></Link>
                        <Link to={`/Login`}><Button>Login</Button></Link>
                        <Link to={`/ViewUsers`}><Button>View Users</Button></Link>

                    </NavItem>
                </Nav>
            </Navbar>
        </div>
    );
}