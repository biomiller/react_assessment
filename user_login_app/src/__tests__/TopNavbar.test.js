import React from 'react';
import renderer from 'react-test-renderer';

import { TopNavbar } from "../components/TopNavbar.js";


it('is navbar rendered', () => {
    const tree=renderer.create(<TopNavbar />).toTree();

    expect(tree).toMatchSnapshot();
});
