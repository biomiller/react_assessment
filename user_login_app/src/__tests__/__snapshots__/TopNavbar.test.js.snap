// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`is TopNavbar rendered 1`] = `
<div>
  <nav
    className="navbar navbar-expand-md navbar-light bg-light"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <a
        className="navbar-brand"
      >
        React Assessment
      </a>
    </a>
    <a
      className="navbar-brand"
    />
    <ul
      className="nav"
    >
      <li
        className="nav-item"
      >
        <a
          href="/Register"
          onClick={[Function]}
        >
          <button
            aria-label={null}
            className="btn btn-secondary"
            onClick={[Function]}
          >
            Register
          </button>
        </a>
        <a
          href="/Login"
          onClick={[Function]}
        >
          <button
            aria-label={null}
            className="btn btn-secondary"
            onClick={[Function]}
          >
            Login
          </button>
        </a>
        <a
          href="/ViewUsers"
          onClick={[Function]}
        >
          <button
            aria-label={null}
            className="btn btn-secondary"
            onClick={[Function]}
          >
            View Users
          </button>
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
